src/main.cpp

```diff
@@ -2457,8 +2464,12 @@ static int64_t nTimeIndex = 0;
 static int64_t nTimeCallbacks = 0;
 static int64_t nTimeTotal = 0;
 
-bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pindex, CCoinsViewCache& view, bool fJustCheck)
+bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pindex, CCoinsViewCache& view, bool fJustCheck, bool fParallel)
```

Comment:
I agree, I don't' like it that much either but IMO it will require a much larger refactor beyond the scope of this PR.  The underlying problem is with Processnewblock, calling AcitvateBestChain,, calling Activatebestchainstep, calling ConnectTip, calling ConnectBlock()...it really all should be in a class (in fact all of main.cpp needs a refactor similar to what Matt did recently for Core) which would make life much easier but I think that should wait until later otherwise this PR  is just going to get bigger than it already is.  I have it in mind to change all this but as I said I think it should wait, and there is nothing that will change regarding the functionality so I think we should leave this for now.